### DISCUSSION TITLE
How to make Booleans from Java look like primitive Javascript booleans?

### DISCUSSION THREAD

**Discussion by jgomer2001** [mozilla/rhino#1139] [Q&A] [answered]
Hi,

I'm starting with Rhino.

Suppose I have the following code:

```
function f() {
   let flag = Packages.com.acme.MyClass.method()
   if (flag){
      //...
   }
}
``` 
Let's say `com.acme.MyClass` is in the classpath and that `method` returns a `java.lang.Boolean`... regardless of true/false returned, the `if` condition in `f` evaluates to true. This is similar to how the Javascript `Boolean` wrapper behaves (see the first example [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean)).

Is there a way to override this behavior so that `java.lang.Boolean`s are treated as JS primitives? I think this is "friendlier"

While I can do it, I'd like to get rid of calling `.booleanValue()`

Any insight appreciated

**Comment by p-bakker** [comment, marked as answer]
Not having looked into this, but I'd say take look at the wrapfactory used by your context and if using the default wrapfactory look at the setting to wrap primitives (or not)

**Reply by tonygermano** [reply]
I think this answer is the correct one if you want them to automatically convert to js primitives.

Alternatively, `if (flag == true)` will also work.
